File: rust/src/models/online_methods_config.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::online_method_config::OnlineMethodConfig;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct OnlineMethodsConfig {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub card: Option<OnlineMethodConfig>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub direct_debit: Option<OnlineMethodConfig>,
}

impl OnlineMethodsConfig {
    pub fn new() -> Self {
        Self {
            card: None,
            direct_debit: None,
        }
    }
}

File: typescript/src/models/onlineMethodsConfig.ts
// this file is @generated
import {
  type OnlineMethodConfig,
  OnlineMethodConfigSerializer,
} from "./onlineMethodConfig";

export interface OnlineMethodsConfig {
  card?: OnlineMethodConfig | null;

  directDebit?: OnlineMethodConfig | null;
}

export const OnlineMethodsConfigSerializer = {
  _fromJsonObject(object: any): OnlineMethodsConfig {
    return {
      card:
        object["card"] != null
          ? OnlineMethodConfigSerializer._fromJsonObject(object["card"])
          : undefined,
      directDebit:
        object["direct_debit"] != null
          ? OnlineMethodConfigSerializer._fromJsonObject(object["direct_debit"])
          : undefined,
    };
  },

  _toJsonObject(self: OnlineMethodsConfig): any {
    return {
      card:
        self.card != null
          ? OnlineMethodConfigSerializer._toJsonObject(self.card)
          : undefined,
      direct_debit:
        self.directDebit != null
          ? OnlineMethodConfigSerializer._toJsonObject(self.directDebit)
          : undefined,
    };
  },
};
